lib/util/cmcd_manager.js

```diff
@@ -55,6 +58,31 @@ shaka.util.CmcdManager = class {
      * @private {boolean}
      */
     this.starved_ = false;
+
+    /**
+     * Attach CMCD data to all HEAD requests so they can be
+     * associated with the playback session.
+     *
+     * @private {shaka.extern.RequestFilter}
+     */
+    this.requestFilter_ = (type, request) => {
+      if (!this.config_.enabled || request.method !== 'HEAD') {
+        return;
+      }
+
+      this.apply_(request);
+    };
+
+    playerInterface.getNetworkingEngine()
```

Comment:
This approach has the problem that there is a publicly exported `networkingEngine.clearAllRequestFilters` method. If a user calls that method, expecting to remove the request filters they had added, it would be bad if it also removed our internal request filters. We haven't used request or response filters internally before this point for that reason.

We make a new CMCD manager (and this a new cmcd header filter) for each asset loaded, so I don't think it would be a common problem, but it could be an issue if someone unloads their request filters mid-playback.

One alternative I can think of (going by your rationale for this design with your comment to Joey, above, to avoid linking the two classes too tightly)...
You could add a separate array of internal-only request filters, which are used in the same way as other request filters but have a non-public register and unregister method.

Reply:
I've refactored the `CmcdManager` and `NetworkingEngine.prototype.request` to pass and optional context to the request call. With this info, CMCD params can be calculated directly as a `RequestFilter`. Additionally an `onRequest` handler has been added to the `NetworkingEngine` constructor which acts as a non-clearable request filter.